swap_meet/vendor.py

```diff
@@ -1,2 +1,196 @@
+        
+        Parameters
+        ----------
+        other_vendor(object): the other vendor that this vendor is trading with
+        my_item(object): an object in this vendor's inventory
+        their_item(object): an object in the other vendor's inventory
+        
+        Returns
+        -------
+        Returns True if the trade successful, False otherwise
+        """
+        if not my_item in self.inventory or not their_item in other_vendor.inventory:
+            return False
+        
+        other_vendor.inventory.append(my_item)
+        self.inventory.remove(my_item)
+
+        self.inventory.append(their_item)
+        other_vendor.inventory.remove(their_item)
+        return True
+
+    
+    def swap_first_item(self, other_vendor):        
+        """
+        Swaps the first item in each inventory between this vendor and the other vendor.
+        
+        Parameters
+        ----------
+        other_vendor(object): the other vendor that this vendor is trading with
+        
+        Returns
+        -------
+        Returns True if the trade was successful, False otherwise
+        """
+        
+        if not self.inventory or not other_vendor.inventory:
+            return False
+            
+        self.swap_items(other_vendor,self.inventory[0],other_vendor.inventory[0])
+        return True
+
+    
+    def get_best_by_category(self, category):
+        """
+        Returns the item with the highest condition value in that specific category.
+        
+        Parameters
+        ----------
+        category(str): category of an inventory item
+        
+        Returns
+        -------
+        Returns the item with highest condition value if successful, None otherwise
+        """
+
+        get_by_category_list = self.get_by_category(category)
+
+        if not get_by_category_list:
+            return None
+
+        best_item = get_by_category_list[0]
+        for item in get_by_category_list:
+            if item.condition > best_item.condition:
+                best_item = item
+        return best_item
+
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        """
+        Swaps the items with the highest condition value in the 
+        preferred category with another vendor.
+        
+        Parameters
+        ----------
+        other_vendor(object): the other vendor that this vendor is trading with
+        my_priority(str): a category requested by this vendor
+        their_priority(str): a category requested by the other vendor
+        
+        Returns
+        -------
+        Returns True if trade was successful, False otherwise
+        """
+
+        my_item = self.get_best_by_category(their_priority)
+        other_item = other.get_best_by_category(my_priority)
+
+        if not my_item or not other_item:
+            return False
+        self.swap_items(other,my_item,other_item)
+        return True
```

Comment:
👍  Woohoo! Good job in using `get_best_by_category` and `swap_items` as helper methods!